conans/test/integration/cache/backup_sources_test.py

```diff
@@ -525,50 +454,33 @@ def source(self):
                    "not found in ['origin', 'http://extrafake/']" in client.out
 
     def test_ok_when_origin_breaks_midway_list(self):
-        client = TestClient(default_server_user=True)
-        download_cache_folder = temp_folder()
-        http_server = StoppableThreadBottle()
-
-        http_server_base_folder_backup1 = temp_folder()
-        http_server_base_folder_backup2 = temp_folder()
+        http_server_base_folder_backup2 = os.path.join(self.file_server.store, "backup2")
 
         sha256 = "315f5bdb76d078c43b8ac0064e4a0164612b1fce77c869345bfc94c75894edd3"
         save(os.path.join(http_server_base_folder_backup2, sha256), "Hello, world!")
         save(os.path.join(http_server_base_folder_backup2, sha256 + ".json"), '{"references": {}}')
 
-        @http_server.server.get("/internet/<file>")
-        def get_internet_file(file):
-            return HTTPError(500, "The server has crashed :( :( :(")
-
-        @http_server.server.get("/downloader1/<file>")
-        def get_file(file):
-            return static_file(file, http_server_base_folder_backup1)
-
-        @http_server.server.get("/downloader2/<file>")
-        def get_file(file):
-            return static_file(file, http_server_base_folder_backup2)
-
-        http_server.run_server()
-
         conanfile = textwrap.dedent(f"""
            from conan import ConanFile
            from conan.tools.files import download
            class Pkg2(ConanFile):
                name = "pkg"
                version = "1.0"
                def source(self):
-                   download(self, "http://localhost:{http_server.port}/internet/myfile.txt", "myfile.txt",
+                   download(self, "{self.file_server.fake_url}/internal_error/myfile.txt", "myfile.txt",
                             sha256="{sha256}")
            """)
 
-        client.save({"global.conf": f"core.sources:download_cache={download_cache_folder}\n"
-                                    f"core.sources:download_urls=['http://localhost:{http_server.port}/downloader1/', "
-                                    f"'origin', 'http://localhost:{http_server.port}/downloader2/']\n"},
-                    path=client.cache.cache_folder)
+        self.client.save(
+            {"global.conf": "tools.files.download:retry=0\n"
```

Comment:
Conan was trying to download it a few times waiting in between, 50% of the time saved comes from this single line :)